Source/WebGPU/WebGPU/BindGroupLayout.mm

```diff
@@ -387,19 +387,19 @@ static void reportErrorInCreateBindGroupLayout(NSString* errorMessage, bool isAu
     }
 
     auto label = fromAPI(descriptor.label);
-    NSArray<MTLArgumentDescriptor *> *argumentDescriptors[stageCount] = { nil, nil, nil };
-    id<MTLArgumentEncoder> argumentEncoders[stageCount];
+    std::array<RetainPtr<NSArray<MTLArgumentDescriptor *>>, stageCount> argumentDescriptors = { nil, nil, nil };
```

Comment:
does `std::array<RetainPtr<T>>` offer benefits over `NSArray<T>` as both will terminate on out of bounds accesses?